src/node.rs

```diff
@@ -247,18 +247,22 @@ impl IrohNode {
         let secret_key = iroh::util::fs::load_secret_key(secret_key_path).await?;
 
         let docs_path = iroh::util::path::IrohPaths::DocsDatabase.with_root(&path);
-        let docs = iroh::sync::store::fs::Store::new(&docs_path)?;
+        let docs = iroh::sync::store::fs::Store::persistent(&docs_path)?;
 
         // create a bao store for the iroh-bytes blobs
-        let blob_path = iroh::util::path::IrohPaths::BaoFlatStoreDir.with_root(&path);
+        let blob_path = iroh::util::path::IrohPaths::BaoStoreDir.with_root(&path);
         tokio::fs::create_dir_all(&blob_path).await?;
-        let db = iroh::bytes::store::flat::Store::load(&blob_path).await?;
-
-        let node = Node::builder(db, docs)
-            .secret_key(secret_key)
-            .spawn()
-            .await?;
-        let sync_client = node.client();
+        let db = iroh::bytes::store::fs::Store::load(&blob_path).await?;
+
+        let node = iroh::node::Builder::with_db_and_store(
+            db,
+            docs,
+            iroh::node::StorageConfig::Persistent(path.clone()),
+        )
+        .secret_key(secret_key)
+        .spawn()
+        .await?;
+        let sync_client = node.clone().client().clone();
 
```

Comment:
I'm concerned about this passage. I needed to `node.clone().client().clone()` to keep the borrow checker happy, because I thing `node.client` returns a reference now?